raspberry-pi/4/poe-hat.nix

```diff
@@ -0,0 +1,122 @@
+{ config, lib, pkgs, ... }:
+
+let 
+  cfg = config.hardware.raspberry-pi."4".poe-hat;
+in {
+  options.hardware = {
+    raspberry-pi."4".poe-hat = {
+      enable = lib.mkEnableOption ''
+        Enable support for the Raspberry Pi POE Hat.
+      '';
```

Comment:
`lib.mkEnableOption` prepends the string "Whether to enable " to the string that is provided as function argument. ([source](https://github.com/NixOS/nixpkgs/blob/master/lib/options.nix#L97))

This way we will get `"Weather to enable Enable support for the ..."`.

I suggest `"support for the Raspberry Pi POE Hat."`